### ISSUE TITLE
Trouble disabling request-body-object

### ISSUE THREAD

**Issue by vmarkovtsev** [IBM/openapi-validator#354]
I think that version 0.52 disabled overriding Spectral rules in `.validaterc` file. I followed [this workaround to disable request-body-object](https://github.com/IBM/openapi-validator/issues/304#issuecomment-958002737) and saw

```
Warning:  'spectral' rules can only be configured through a valid Spectral config file.
Spectral rule settings in the .validaterc file are ignored.
```

So I read the documentation and created `.spectral.yaml` file in the same directory as `.validaterc`. The content was:

```
extends: @ibm-cloud/openapi-ruleset
rules:
  request-body-object: off
```

However, it didn't do the trick and I still hit `request-body-object` rule errors. Ditto with `--ruleset .spectral.yaml`. I wonder if I am missing something or there is a bug.

**Comment by vmarkovtsev** [comment]
I found that the following Spectral config works:

```
extends: '@ibm-cloud/openapi-ruleset'
rules:
  request-body-object: off
```

Hence my problem was with the single quotes. I didn't have them because I copy-pasted from the documentation.

My issue transforms to the linter not reporting YAML validation errors in `.spectral.yaml` and silently ignoring the config. Version 0.51.3 reports them.

**Comment by dpopp07** [comment]
@vmarkovtsev Documentation is now fixed, thanks to your PR. As for YAML validation errors, we no longer do the parsing but let Spectral handle it. We should be printing any Spectral errors we receive if there is an issue - are you seeing otherwise?

**Comment by vmarkovtsev** [comment]
@dpopp07 As far as I see Spectral stays silent on parsing errors.

**Comment by jnnyrttn** [comment]
For me using `extends: '@ibm-cloud/openapi-ruleset'` with  `-r .spectral.yaml` parameter still does not work. 

```
[Info] Problem reading Spectral config file, using default IBM Spectral ruleset. Cause for error:
ENOENT: no such file or directory, open '/home/*/*/*/*/@ibm-cloud/openapi-ruleset'
```

It only works when i do somehing like this:
```javascript
extends: /path/to/node_modules/@ibm-cloud/openapi-ruleset/src/ibm-oas.js
rules:
 some-rule: 'off'

**Comment by vmarkovtsev** [comment]
I hit the same error as @jnnyrttn in https://github.com/IBM/openapi-validator/issues/366

**Comment by jnnyrttn** [comment]
@dpopp07 After installing 0.54.1 i hit the same error. I see old versions or spectral-*.  in package.json

```
    "@stoplight/spectral-cli": "^6.1.0",
    "@stoplight/spectral-core": "^1.6.1",
```
I am using: 

node v17.4.0
npm 8.3.1

**Comment by dpopp07** [comment]
@jnnyrttn Thanks for checking that. The latest version of the validator (`0.56.2`) includes pinned Spectral dependencies in the `package.json`. That has solved the issue for me. Let me know if you're still experiencing the issue in that version. If you are, I wonder if old versions are being cached somewhere.

**Comment by DanRoseus** [comment]
I also see this when using the latest docker image to run. I.e. run using the following:
```sh
docker run --volume `pwd`:/data jamescooke/openapi-validator:0.57.0 swagger.yaml 
```

The following .spectral.yaml seems to be ignored:
```yaml
extends: '@ibm-cloud/openapi-ruleset'
rules:
  property-case-convention: off
```

And the following works:
```yaml
extends: '/usr/local/lib/node_modules/ibm-openapi-validator/node_modules/@ibm-cloud/openapi-ruleset/src/ibm-oas.js'
rules:
  property-case-convention: off
```

**Comment by dpopp07** [comment]
@jnnyrttn @DanRoseus I am seeing this in the latest version of the validator as well. I don't think that pinning dependencies made a difference (my testing environment must have been messed up somehow).

This is a bug in Spectral and I've opened an issue against their repository - https://github.com/stoplightio/spectral/issues/2072

I'll note a workaround here - if you install the `@ibm-cloud/openapi-ruleset` package in a local folder and run the tool from that folder, Spectral is able to find the ruleset in the file system. Not ideal but hopefully that helps in the meantime.

**Comment by dpopp07** [comment]
@vmarkovtsev @jnnyrttn @DanRoseus To keep you updated on the latest info from Spectral, it turns out that it's actually _required_ for rulesets to live in the local file system. See [this comment on the issue I opened](https://github.com/stoplightio/spectral/issues/2072#issuecomment-1047630488). So, it turns out the workaround I mentioned is actually just the standard use case. So you'll need to install the `@ibm-cloud/openapi-ruleset` in the same directory that your Spectral config file lives in.

I will update our documentation to reflect this and will close this issue once I do. If any of y'all run into issues using the prescribed method, post here and I can re-open.

**Comment by smridge** [comment]
I'm having this issue as well. I can't for the life of me get a handful of rules ignored for the latest version `0.81.0`..the most notable being:
- `property-case-convention`
- `oas3-schema`

**Comment by dpopp07** [comment]
@smridge Can you share more about your configuration? This should now be working [as documented](https://github.com/IBM/openapi-validator/blob/main/docs/ibm-cloud-rules.md#customization)

**Comment by smridge** [comment]
> @smridge Can you share more about your configuration? This should now be working [as documented](https://github.com/IBM/openapi-validator/blob/main/docs/ibm-cloud-rules.md#customization)

Ah ok, I will give this another try and see if I still run into any issues. Thank you for the reference!

**Comment by flowstef** [comment]
This has been troubling me for a while now and I tried again today with v0.96.0 but unless I use @DanRoseus's workaround, my `.spectral.yaml` is ignored.

**Comment by dpopp07** [comment]
Are you seeing that the `.spectral.yaml` file is being ignored altogether or that it's not being honored because the ruleset it's pointing to can't be found?

The location of the ruleset in the filesystem relative to the tool does matter

**Comment by flowstef** [comment]
Good question, I don't have the answer to that.
I'm using the docker version of the tool and run it like this:
`docker run --volume "$(PWD)":/data jamescooke/openapi-validator:0.96.0 -v -s -r .spectral.yaml ./src/api.json`
My guess is that it's the latter but I'm not sure how to debug it - the docker logs don't show any errors.

**Comment by dpopp07** [comment]
Oh okay, I see. It looks like it is a matter of not finding the file, since the docker image probably doesn't include the ruleset as a separate package. I know it's a little hacky but I think you can workaround this by pointing the config file directly at the ruleset dependency. Something like `extends: <path>/node_modules/@ibm-cloud/openapi-ruleset/src/ibm-oas.js`

**Comment by flowstef** [comment]
Yep, that's what I'm doing. I just thought I'd comment to show this is still an issue (even though the ticket is closed) so others can find [the solution](https://github.com/IBM/openapi-validator/issues/354#issuecomment-1044311937) more easily.

**Comment by dpopp07** [comment]
I appreciate it. I'm leaving it closed because it seems the issue is not with this project but with the Docker image, which is community-supported.

@jamescooke if you're still maintaining the Docker image, perhaps the image should include the ruleset package as a dependency so the config file can specify `extends: '@ibm-cloud/openapi-ruleset'` without any trouble.

**Comment by jamescooke** [comment]
@dpopp07 Yes, I'm still maintaining the Docker image, but I'm no longer using the validator on a daily basis at work.

I've opened https://github.com/jamescooke/openapi-validator-docker/issues/200

There are specific questions on this image issue - TL;DR:

* What's the smallest failing test that can be built into the GitHub Actions system to ensure that the ruleset has been installed correctly?
* Does the docker image build system need to track the version / tag of `openapi-ruleset` ?

All answers / support very welcome - please add them to https://github.com/jamescooke/openapi-validator-docker/issues/200
